sklearn/utils/estimator_checks.py

```diff
@@ -1028,9 +1028,9 @@ def check_array_api_input_and_values(
 def _check_estimator_sparse_container(name, estimator_orig, sparse_type):
     rng = np.random.RandomState(0)
     X = rng.uniform(size=(40, 3))
-    X[X < 0.8] = 0
+    X[X < 0.6] = 0
```

Comment:
This fixes the problem because with `X<0.6` not all elements are set to zero, but with `<0.8` that does happen?

Reply:
The probability of getting an all-zero training set when subsampling in Halving*SearchCV is lower with the 0.6 threshold. Tests pass locally.